source/include/shadow.h

```diff
@@ -265,362 +307,342 @@ typedef enum ShadowStatus
  * A shadow topic string takes one of the two forms, in the case of an unnamed ("Classic") shadow:
  *     $aws/things/\<thingName\>/shadow/\<operation\>
- *     $aws/things/\<thingName\>/shadow/\<operation\>/\<suffix\>
  *
  * Or as follows, in the case of a named shadow:
- *     $aws/things/\<thingName\>/shadow/name/<shadowName\>/<operation\>
- *     $aws/things/\<thingName\>/shadow/name/<shadowName\>/<operation\>/\<suffix\>
+ *     $aws/things/\<thingName\>/shadow/name/\<shadowName\>/\<operation\>
  *
- * The \<thingName\>, \<shadowName\>, \<operation\> and \<suffix\> segments correspond to the
- * four input parameters of this macro. The \<suffix\> part can be null.
+ * The \<thingName\>, \<shadowName\> and \<operation\> segments correspond to the
+ * three input parameters of this macro.
  *
- * When thingName and shadow name are known to be "myThing"  and "myShadow" at compile time,
+ * When thingName and shadow name are known to be "myThing" and "myShadow" at compile time,
  * invoke the macro like this:
  * (In this case, the length is a constant at compile time.)
  *
- *     SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_LENGTH, SHADOW_SUFFIX_DELTA_LENGTH, 7, 8 )
+ *     SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_DELTA_LENGTH, 7, 8 )
  *
  * When thingName and shadowName are only known at run time, and held in variables myThingName
  * and myShadowName, invoke the macro like this:
  *
- *     SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_LENGTH, SHADOW_SUFFIX_DELTA_LENGTH,
- *                       strlen( ( const char * ) myThingName ),
- *                       strlen( ( const char * ) myShadowName ) )
+ *     SHADOW_TOPIC_LENGTH( SHADOW_OP_UPDATE_DELTA_LENGTH,
+ *                          strlen( ( const char * ) myThingName ),
+ *                          strlen( ( const char * ) myShadowName ) )
  *
- * To use an unnamed ("Classic") shadow, the shadowName length passed must be zero.
+ * To compute an unnamed ("Classic") shadow length, the shadowName length passed must be zero.
  *
  * @param[in] operationLength   Can be one of:
  *                                  - #SHADOW_OP_UPDATE_LENGTH
- *                                  - #SHADOW_OP_DELETE_LENGTH
+ *                                  - #SHADOW_OP_UPDATE_ACCEPTED_LENGTH
+ *                                  - #SHADOW_OP_UPDATE_REJECTED_LENGTH
+ *                                  - #SHADOW_OP_UPDATE_DOCUMENTS_LENGTH
+ *                                  - #SHADOW_OP_UPDATE_DELTA_LENGTH
  *                                  - #SHADOW_OP_GET_LENGTH
- * @param[in] suffixLength      Can be one of:
- *                                  - #SHADOW_SUFFIX_NULL_LENGTH
- *                                  - #SHADOW_SUFFIX_ACCEPTED_LENGTH
- *                                  - #SHADOW_SUFFIX_REJECTED_LENGTH
- *                                  - #SHADOW_SUFFIX_DELTA_LENGTH
- *                                  - #SHADOW_SUFFIX_DOCUMENTS_LENGTH
+ *                                  - #SHADOW_OP_GET_ACCEPTED_LENGTH
```

Comment:
Same comment as the param naming. These are not operations. Rename the param?

Reply:
Un-done by new commit today.